internal/pkg/cli/task_delete.go

```diff
@@ -262,26 +281,98 @@ func (o *deleteTaskOpts) askTaskName() error {
 		if err != nil {
 			return fmt.Errorf("select task from default cluster: %w", err)
 		}
-		o.name = task.Name
+		o.name = task
 		return nil
 	}
 	task, err := sel.Task(taskDeleteNamePrompt, "", selector.TaskWithAppEnv(o.app, o.env))
 	if err != nil {
 		return fmt.Errorf("select task from environment: %w", err)
 	}
-	o.name = task.Name
+	o.name = task
 	return nil
 }
 
 func (o *deleteTaskOpts) Execute() error {
 	// Get clients.
+	sess, err := o.getSession()
+	if err != nil {
+		return fmt.Errorf("get session: %w", err)
+	}
+
+	ecsClient := o.newTaskListerStopper(sess)
+	taskDeleter := o.newTaskDeleter(sess)
+
+	// ECR Deletion happens from the default profile in app delete. We can do it here too.
+	defaultSess, err := o.sess.DefaultWithRegion(aws.StringValue(sess.Config.Region))
+	if err != nil {
+		return fmt.Errorf("get default session for ECR deletion: %s", err)
+	}
+	ecrDeleter := o.newImageRemover(defaultSess)
+
+	// Get information about the task stack. This struct will be used to get the names of the ECR
+	// repo and task stack.
+	taskInfo, err := taskDeleter.GetTaskStack(o.name)
+	if err != nil {
+		return fmt.Errorf("retrieve stack information for task %s: %w", o.name, err)
+	}
+
+	o.spinner.Start(fmt.Sprintf("Cleaning up resources for task %s.", color.HighlightUserInput(o.name)))
+	// Get running tasks in family.
+	var tasks []*awsecs.Task
+	if o.defaultCluster {
+		tasks, err = ecsClient.ListActiveDefaultClusterTasks(ecs.ListTasksFilter{
+			TaskGroup: o.name,
+		})
+		if err != nil {
+			o.spinner.Stop(log.Serrorln("Error listing running tasks."))
+			return fmt.Errorf("list running tasks in default cluster: %w", err)
+		}
```

Comment:
I feel `StopDefaultClusterTasks` doesn't abstract enough logic. We probably don't need to list active task here but embed the logic in `StopDefaultClusterTasks` so that `StopDefaultClusterTasks` takes the filter:

```suggestion
		tasks, err = ecsClient.StopDefaultClusterTasks(ecs.StopTasksFilter{
			TaskGroup: o.name,
		})
		if err != nil {
			o.spinner.Stop(log.Serrorln("Error stopping running tasks."))
			return fmt.Errorf("stop running tasks in default cluster: %w", err)
		}
```
What do you think?

Reply:
I've gone even further than this and embedded the listing logic in several new top level functions in the `ecs` package: 
* `StopDefaultClusterTasks(familyName string)` which stops one-off copilot tasks in the default cluster.
* `StopAppEnvOneOffTasks(app, env, family string)` which stops one-off copilot tasks in the given environment.
* `StopActiveWorkloadTasks(app, env, workload string)` which stops all copilot tasks from the given workload and env. 
These abstractions are easier to work with and mock, and I've added them in `job delete` and `task delete` where applicable. 

The `List` methods are still used by the `selector` package but are not needed in `cli`, so I've removed them from `interfaces.go`.